sub-projects/device/src/aws.py

```diff
@@ -0,0 +1,42 @@
+import boto3
+import json
+from PIL import Image
+
+class Rekgoniton(object):
+	"""
+	A class that creates a client object for access to AWS Rekgoniton.
+	"""
+	def __init__(self):
+		self.client = boto3.client("rekognition")
+		self.image_list = ['cameron.jpg','rachel.jpg']
+
+	def compare(self, img):
+		"""
+		Compares a face from the S3 bucket to a passed in image represented in
+		bytes.
+		:param img: target image for comparing
+		"""
+		i = 0
+		while i < len(self.image_list):
+		    response = self.client.compare_faces(
```

Comment:
This will get assigned a new value on each iteration of the loop. Meaning that if on `i=0` it finds a match, it will get overwritten on any subsequent values of `i`. This alone is not the problem, but combined with the logic following the loop it means that the result will always be if the given face matches the last face in `self.image_list`.

Reply:
I am not sure I understand. Are you talking about the response variable?

Reply:
Yes, because it is not ever used until the while loop is finished.